lib/udt.js

```diff
@@ -53,20 +53,35 @@ class Point {
   }
 }
 
-const parsePoints = (buffer, count) => {
+const parsePoints = (buffer, count, isGeometryPoint) => {
   // s2.1.5 + s2.1.6
+  // The key distinction for parsing is that a GEOGRAPHY POINT is ordered Lat (y) then Long (x),
+  // while a GEOMETRY POINT is ordered x then y.
+  // Further, there are additional range constraints on GEOGRAPHY POINT that are useful for testing that the coordinate order has not been flipped, such as that Lat must be in the range [-90, +90].
 
   const points = []
   if (count < 1) {
     return points
   }
 
-  for (let i = 1; i <= count; i++) {
-    const point = new Point()
-    points.push(point)
-    point.x = buffer.readDoubleLE(buffer.position)
-    point.y = buffer.readDoubleLE(buffer.position + 8)
-    buffer.position += 16
+  if (isGeometryPoint) {
+    // GEOGRAPHY POINT (s2.1.6): x then y.
+    for (let i = 1; i <= count; i++) {
+      const point = new Point()
+      points.push(point)
+      point.x = buffer.readDoubleLE(buffer.position)
+      point.y = buffer.readDoubleLE(buffer.position + 8)
+      buffer.position += 16
+    }
+  } else {
+    // GEOGRAPHY POINT (s2.1.5): Lat (y) then Long (x).
+    for (let i = 1; i <= count; i++) {
+      const point = new Point()
+      points.push(point)
+      point.y = buffer.readDoubleLE(buffer.position)
```

Comment:
I'm wondering if for geography points it's worth either naming them lat/lng or adding a duplicate props of lat/lng. What do you think?

Reply:
that's an interesting idea. and it avoids a lot of confusion.

as a bonus, we could leave the x/y flipped as before, and include correct lat/lng alongside, and then i believe it would no longer be a breaking change?

Reply:
(and then possibly also do a follow-up breaking change that updates or eliminates the x/y in favor of lat/lng, but that would enable the fix landing in things that can't take a breaking change just now.)

Reply:
great idea

Reply:
ok, i've addressed the other issues. i will need to do a bit more work for this other direction, but i like it. :)